### PR TITLE
add juju abort command 

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description of change

Add the juju abort command for branches and related tests.  Includes an update
the modelgeneration package in api.

## QA steps

1. export JUJU_DEV_FEATURE_FLAGS=generations
2. juju bootstrap
2. juju deploy wordpress
3. juju add-branch apple
2. juju track apple wordpress 
2. juju abort apple             <- fails due to assigned units.
4. juju commit apple
3. juju add-branch banana
3. juju config wordpress debug=yes --branch banana
3. juju abort banana     <- succeeds

Verify in the juju db,"generation-id" : 0, and completed and completed-by are set.

Please try other permutations as well.
## Documentation changes

yes

### CODE REVIEW

File: api/modelgeneration/modelgeneration_test.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func (s *modelGenerationSuite) setUpMocks(c *gc.C) *gomock.Controller {
 	return ctrl
 }
 
-func (s *modelGenerationSuite) TestAddGeneration(c *gc.C) {
+func (s *modelGenerationSuite) TestAddBranch(c *gc.C) {
```
Comment (hmlanigan) [inline api/modelgeneration/modelgeneration_test.go]: drive-by

File: api/modelgeneration/modelgeneration_test.go [hunk 2]
```diff
@@ -112,7 +124,7 @@ func (s *modelGenerationSuite) TestHasActiveBranch(c *gc.C) {
 	c.Check(has, jc.IsTrue)
 }
 
-func (s *modelGenerationSuite) TestGenerationInfo(c *gc.C) {
+func (s *modelGenerationSuite) TestBranchInfo(c *gc.C) {
```
Comment (hmlanigan) [inline api/modelgeneration/modelgeneration_test.go]: drive by

**Comment by hmlanigan** [comment]
unit test failures in modelGenerationSuite.TestHasActiveBranch* have been fixed in 10388.

**Comment by hmlanigan** [comment]
force push to rebase the branch now that pr 10388 has landed.  no changes to code under review.

**Comment by mitechie** [comment]
Awesome, thanks for this Heather. Looks great, couple of comments. 

QA:

abort with a tracking unit:

    ERROR branch has assigned units

suggestion

    ERROR branch is in progress. Either reset values on tracking units or remove them to abort.

I tried to remove the unit that was tracking so I could abort but due to https://bugs.launchpad.net/juju/+bug/1834553 I could not. 

Due to this I can't finish really QA'ing the abort process once we get units tracking and try to remove them. 

I did verify that if I change the config values back to the original (on another branch/charm change) I could abort it successfully. 

Looks really good otherwise. Thank you!

**Comment by hmlanigan** [comment]
used --force instead of commit for output change requested after QA, then squash.

$$merge$$

**Comment by hmlanigan** [comment]
known unrelated intermittent unit test failure

$$merge$$
